### ISSUE TITLE
"Activating extension" announced all the time by screen reader

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#95357]
from @marlon-sousa:

The activating extension message is being read automatically by the screen readers while we are on the editor or at any other part. This became more prominent in the last release version of vscode and is taking productivity down in a way which is making the use of the software not viable at all, specially when you use extensions that have tasks in background.

My experience shows that, whenever an extension is doing anything, this message is being output to screen readers.

For example, the indent nave vscode extension (https://github.com/mltony/vscode-indent-nav) is used by me to quickly jump from a line with a given indentation level to the next / previous line with the same indentation level, which allows me to quickly navigate by method or class.
Each time I press alt + down arrow (thus issuing a command defined by this extension) I listen to the activating extension phrase.

**Comment by isidorn** [comment]
I looked into this and the issue here is that the Status bar now has the "status" role. However we explicitly set the `aria-live="off"` on that element, but VoiceOver and NVDA still read the changes in the status bar as if the `aria-live` was set to "polite".
I tried setting `aria-live` to `off` to aslo the children of the status bar container but to no avail.

I see that orca is behaving well here, and that status bar changes are not read out. @joanmarie
Might it be that this is an issue in NVDA @leonardder
If yes, we might have to file it also to VoiceOver and until this gets fixed we might set the statusbar role to "status" only on Linux.

fyi @marcozehe

**Comment by marlon-sousa** [comment]
@isidorn , the user to first report this at gitter was @jvesouza. The system he uses is linux and orca is his screen reader.

If orca is also repeating the activating extension, them it seems that your conclusions might be wrong. @jvesouza can you confirm that this message is being spoken on Liniux with Orca?

**Comment by jvesouza** [comment]
@isidorn @marlon-sousa I'm not sure the problem is the same, but from time to time I hear the orca announce the following:
info activate extensions occured NN times.

**Comment by joanmarie** [comment]
I know that Orca is not speaking live regions from the status bar. I can see those in my debug.out, where Orca recognizes these events as live regions, but then doesn't present them because of `aria-live` being set to off.

@jvesouza: If you can capture a debug.out where Orca says the "occured NN times", I should be able to provide the element it came from.

**Comment by jvesouza** [comment]
@joanmarie I sent a debug file with two occurrences of the message.

**Comment by joanmarie** [comment]
@jvesouza: Thanks!

@marlon-sousa: What José is experiencing is *not* coming from the status bar; it's coming from a `div` with an ARIA role of `alert`, class of `monaco-alert`.

**Comment by leonardder** [comment]
Alerts are automatically spoken by NVDA when they're added to the DOM and become visible, as far as I know.

**Comment by MarcoZehe** [comment]
Correct, and `role="alert"` has an implicit `aria-live="assertive"` attached to it. Alerts are things that really *need* to be spoken and grab the user's attention.

**Comment by MarcoZehe** [comment]
The element this is set on is not part of the status bar. I checked, and aria-live="off" gets properly propagated to those status bar elements. The role "alert" is on an element with class monaco-alert. And this sounds appropriate to have an alert, since this usually contains critical error messages or hints that mean the user should take care of it.

**Comment by isidorn** [comment]
Thanks for investigating, it seems like somebody is explicitly calling `alert`. I will fix this very soon.

**Comment by isidorn** [comment]
We have a concept of silent notifications which should only appear in the status bar and not cause a big ruckus. However there was an issue that for silent notifications we were using aria alerts. I have pushed a commit which fixes that.

Please note that we still use aria alerts for non silent notifications. Do let me know if you hate that or if you have ideas how we can improve there.

fyi @bpasero

**Comment by leonardder** [comment]
I wonder how we could make these silent notifications opt in in such a way that they're less obnoxious.

**Comment by isidorn** [comment]
@leonardder now they will not be read out. Who wants to get them should Focus on the status bar or F1 > show notifications. So now we are good regardign silent notifications. My question is more directed towards all other notifications. Should we mute them as well (for now I would not).

**Comment by leonardder** [comment]
> Should we mute them as well (for now I would not).

Nope, I wouldn't mute them either. The question is whether they are important enough to be aria live assertive. May be polite would make them less obnoxious. Thoughts @MarcoZehe?

**Comment by MarcoZehe** [comment]
No, I don't think we should mute all notifications. The ones that are notifications and are not silent should definitely be read.

**Comment by isidorn** [comment]
Yes we can definetly change them to be polite.
To be honest we have both apis, to have notifications `polite` and `assertive` but clients seem to just use `assertive` all the time. I could change all to be `polite`.

**Comment by jvesouza** [comment]
Bye bye activating extension!
After updating the VSCode this message didn't reach my ears anymore! That is: problem fixed
Thank you all.

**Comment by isidorn** [comment]
@jvesouza awesome! Adding verified label.
